src/components/_classes/list/ListComponent.js

```diff
@@ -0,0 +1,226 @@
+        console.warn(err.message);
+      }
+    }
+
+    return headers;
+  }
+
+  getOptionTemplate(data, value) {
+    if (!this.component.template) {
+      return data.label;
+    }
+    const options = {
+      noeval: true,
+      data: {}
+    };
+    const template = this.sanitize(
+      this.component.template
+        ? this.interpolate(this.component.template, { item: data })
+        : data.label,
+      this.shouldSanitizeValue,
+    );
+    if (value && !_.isObject(value) && options.data.item) {
+      // If the value is not an object, then we need to save the template data off for when it is selected.
+      this.templateData[value] = options.data.item;
+    }
+    return template;
+  }
+
+  itemTemplate(data, value) {
+    if (_.isEmpty(data)) {
+      return '';
+    }
+
+    const template = this.sanitize(this.getOptionTemplate(data, value), this.shouldSanitizeValue);
+    if (template) {
+      const label = template.replace(/<\/?[^>]+(>|$)/g, '');
+      const hasTranslator = this.i18next?.translator;
+      if (!label || (hasTranslator && !this.t(label, { _userInput: true }))) return;
+      return hasTranslator ? template.replace(label, this.t(label, { _userInput: true })) : label;
+    }
+    else {
+      return this.sanitize(JSON.stringify(data), this.shouldSanitizeValue);
+    }
+  }
+
+  handleLoadingError(err) {
+    this.loading = false;
+    if (err.networkError) {
+      this.networkError = true;
+    }
+    this.itemsLoadedResolve();
+    this.emit('componentError', {
+      component: this.component,
+      message: err.toString(),
+    });
+    console.warn(`Unable to load resources for ${this.key}`);
+  }
+
+  /* eslint-disable max-statements */
+  updateItems(searchInput, forceUpdate) {
+    if (!this.component.data) {
+      console.warn(`Select component ${this.key} does not have data configuration.`);
+      this.itemsLoadedResolve();
+      return;
+    }
+
+    // Only load the data if it is visible.
+    if (!this.visible) {
+      this.itemsLoadedResolve();
+      return;
+    }
+
+    switch (this.component.dataSrc) {
+      case 'values':
+        this.setItems(this.component.data.values);
+        break;
+      case 'json':
+        this.setItems(this.component.data.json);
+        break;
+      case 'custom':
+        this.updateCustomItems(forceUpdate);
+        break;
+      case 'resource': {
+        // If there is no resource, or we are lazyLoading, wait until active.
+        if (!this.component.data.resource || (!forceUpdate && !this.active)) {
+          this.itemsLoadedResolve();
+          return;
+        }
+
+        let resourceUrl = this.options.formio ? this.options.formio.formsUrl : `${Formio.getProjectUrl()}/form`;
+        resourceUrl += (`/${this.component.data.resource}/submission`);
+
+        if (forceUpdate || this.additionalResourcesAvailable || !this.serverCount) {
+          try {
+            this.loadItems(resourceUrl, searchInput, this.requestHeaders);
```

Comment:
ListComponent does not have its own or prototype loadItems method. That is not correct. If someone inherits from ListComponent class and has no idea what methods it is expected to have, it will break everything. We need to add its own loadItems method that will do, at least, nothing and comment it like 'Must be implemented in child classes'.
The same is for setItems method.

Reply:
Methods added. All tests passed locally